packages/drivers/odsp-driver/src/getFileLink.ts

```diff
@@ -95,6 +105,48 @@ export async function getFileLink(
 	return fileLink;
 }
 
+/**
+ * Handles location redirection while fulfilling the getFilelink call. We don't want browser to handle
+ * the redirection as the browser will retry with same auth token which will not work as we need app
+ * to regenerate tokens for the new site domain.
+ * @param getToken - token fetcher to fetch the token.
+ * @param odspUrlParts - parts of odsp resolved url.
+ * @param logger - logger to send events.
+ * @returns Response from the API call.
+ * @alpha
+ */
+async function getFileLinkWithLocationRedirectionHandling(
+	getToken: TokenFetcher<OdspResourceTokenFetchOptions>,
+	odspUrlParts: IOdspUrlParts,
+	logger: ITelemetryLoggerExt,
+): Promise<string> {
+	// We can have chains of location redirection one after the other, so have a for loop
+	// so that we can keep handling the same type of error.
+	for (;;) {
+		try {
+			return await getFileLinkCore(getToken, odspUrlParts, logger);
+		} catch (error: unknown) {
+			if (
+				isFluidError(error) &&
+				error.errorType === OdspErrorTypes.fileNotFoundOrAccessDeniedError
+			) {
+				const redirectLocation = (error as IOdspErrorAugmentations).redirectLocation;
+				if (redirectLocation !== undefined) {
+					logger.sendTelemetryEvent({
+						eventName: "LocationRedirectionErrorForGetOdspFileLink",
```

Comment:
I know we can't use URIs in payload, but it would be great to have something that helps us
1. See in telemetry if we have a chain of redirects (maybe an iteration counter)
2. maybe track individual instances (like hash of domain, such that we can't get to actual domain value, but we can say - that one instance is responsible for N% of all redirects)

#2 is probably not needed unless we learn otherwise.

Reply:
Added prop in telemetry eevnt.